docs/blueprints/referral-manager-v1.md

```diff
@@ -0,0 +1,236 @@
+# Referral Manager V1: Cross-Satellite Referral Link
+
+## Abstract
+
+This document describes how to handle user referrals across satellites.
+
+## Background
+
+Previous referral program only works per satellite. This means a user on us-east-1 satellite can only refer people to join us-east-1.
+
+We want referrals to be valid on all Tardigrade satellites. This will allow the user to select the satellite that is best for them.
+
+### Non-Goals
+A registration page where a user can select their preferred Satellite during registration.
+
+## Design
+
+### Definitions
+
+Referral Manager
+: A stand-alone process running outside of satellites. It should only talk to tardigrade branded satellites.
+
+Referral Manager CLI
+: A CLI that will allow operators to invoke referral link generation.
+
+Referral Link
+: A one-time-use URL that contains a unique invitation token.
+
+Invitation Token
+: A random number generated by Referral Manager for each user in the referral program.
+
+Eligible Users
+: Users who have 0 unredeemed referral link.
+
+Owner's Satellite
+: The satellite that an owner of a particular token belongs to.
+
+Redeemed Satellite
+: The satellite where the new user redeeming a token has registered on.
+
+### How it will work
+
+_User Interface_
+
+1. Users will have a referral tab on the UI. On click, it will trigger satellite to send a request to retrieve referral links from Referral Manager:
```

Comment:
Will it do this on every request? Is there a debounce? Is there a cache?

Reply:
As if right now, yes, it's happening on every request so the data is up-to-date. We probably should have a debounce. Maybe we can say, it should only make one request per hour?